src/main/java/com/bootcamp/demo/business/AccountValidation.java

```diff
@@ -0,0 +1,115 @@
+                validationResponse.addMessage("\nWrong email format");
+        }
+    }
+
+    public void usernameValidator() {
+        if (StringUtils.isEmpty(account.getUsername()))
+            validationResponse.addMessage("\nUsername field is empty");
+        else {
+            final String userNameExpression = "[a-zA-Z0-9]*$";
+            if (!account.getUsername().matches(userNameExpression))
+                validationResponse.addMessage("\nWrong username format: must contain only digits and letter ");
+        }
+    }
+
+    public void passwordValidator() {
+        if (StringUtils.isEmpty(account.getPassword()))
+            validationResponse.addMessage("\nPassword field is empty");
+    }
+
+    public void phoneNumberValidator() {
+        if (StringUtils.isEmpty(account.getPhoneNumber()))
+            validationResponse.addMessage("\nPhoneNumber field is empty");
+        else if (account.getPhoneNumber().length() != 10)
+            validationResponse.addMessage("\nWrong phoneNumber format: must have 10 digits ");
+        else {
+            boolean match = Arrays.asList(account.getPhoneNumber().split(""))
+                    .stream()
+                    .allMatch(str -> str.matches("\\d+"));
+            if (!match)
+                validationResponse.addMessage("\nWrong phoneNumber format: must contain only digits ");
+        }
+    }
+
+    private void addressValidator() {
+        if (StringUtils.isEmpty(account.getAddress()))
+            validationResponse.addMessage("\nAddress field is empty");
+    }
+
+    private void dateOfBirthValidator() {
+        if (StringUtils.isEmpty(account.getDateOfBirth()))
+            validationResponse.addMessage("\nDateOfBirth field is empty");
+        else if (account.getDateOfBirth().length() != 10)
+            validationResponse.addMessage("\nWrong dateOfBirth format: must be MM-dd-yyyy ");
+        else {
+            String pattern = "MM-dd-yyyy";
+            SimpleDateFormat simpleDateFormat = new SimpleDateFormat(pattern);
+            try {
+                Date date = simpleDateFormat.parse(account.getDateOfBirth());
+                LocalDate localDate = date.toInstant().atZone(ZoneId.systemDefault()).toLocalDate();
+                Period period = Period.between(localDate, LocalDate.now());
+                if (period.getYears() < 18)
+                    validationResponse.addMessage("\nMinimum age must be 18 ");
+            } catch (ParseException e) {
+                validationResponse.addMessage("\nWrong dateOfBirth format: must be MM-dd-yyyy ");
+                e.printStackTrace();
+            }
+        }
+    }
+
+    public void validate() {
+        nameValidator();
+        usernameValidator();
+        emailValidator();
+        passwordValidator();
+        phoneNumberValidator();
+        addressValidator();
+        dateOfBirthValidator();
+
+        if (!validationResponse.getMessages().isEmpty())
```

Comment:
is this logic correct? Shouldn't the validate status be valid if there are no validation error messages present?